utils/iio_common.c

```diff
@@ -331,6 +331,9 @@ struct iio_context * handle_common_opts(char * name, int argc,
 				goto err_fail;
 			}
 			params.timeout_ms = sanitize_clamp("timeout", optarg, 0, INT_MAX);
+			/* Wait forever if timeout is explicitly set to 0 */
+			if (!params.timeout_ms)
+				params.timeout_ms = -1;
```

Comment:
But what happens if no `--timeout` option is given? If that defaullts to the backend default, then it makes sense. However, if it defaults to "wait forever", I think it would make sense to update the meaning of the argument to match the `params.timeout_ms` meaning.